### PR TITLE
Upgrade to rails 5 and dependencies

### PR DESCRIPTION
**Pull request by gigamorph**

### CODE REVIEW

File: config/cable.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+development:
+  adapter: async
+
+test:
+  adapter: async
+
+production:
+  adapter: redis
+  url: redis://localhost:6379/1
```
Comment (hshyk) [inline config/cable.yml]: should this be hardcoded?

Comment (gigamorph) [reply config/cable.yml]: This file has been added automatically by the update script. It contains configuration for "Action Cable", a pretty new feature using WebSockets and I'm sure we are not using it and the redis URL specified here is irrelevant.

File: config/environments/development.rb [hunk 2]
```diff
@@ -3,19 +3,32 @@
 
   # In the development environment your application's code is reloaded on
   # every request. This slows down response time but is perfect for development
-  # since you don't have to restart the web server when you make code changes.
+  # since you don't have to resag the web server when you make code changes.
```
Comment (mr-sigma) [inline config/environments/development.rb]: Typo?

File: spec/controllers/annotation_layers_controller_spec.rb [hunk 3]
```diff
@@ -3,41 +3,46 @@
...
+          post_to_create(@layer)
           @annotationLayer = AnnotationLayer.last()
           expect(@annotationLayer['version']).to eq (1)
         end
 
         it 'does not fail validation if otherContent is nil' do
-          json = JSON.parse(@annoLayerString)
-          json['otherContent'] = nil
-          post :create, json
+          layer = @layer
```
Comment (mr-sigma) [inline spec/controllers/annotation_layers_controller_spec.rb]: Isn't this line unnecessary? You're modifying layer in the line below which should change `@layer` and then it just gets undone again before the next test.

Do we need an `@layer.clone` or the equivalent?

File: spec/controllers/annotation_lists_controller_spec.rb [hunk 4]
```diff
@@ -3,223 +3,61 @@
...
+    @annotationLayer = AnnotationLayer.create_from_iiif(layer)
   end
 
   context 'when Get is called' do
     describe 'GET annotation json' do
       before(:each) do
         AnnotationList.delete_all
-        layer_obj = IIIF::Layer.create(id: '/layer/1')
-        AnnotationLayer.create_from_iiif(layer_obj)
+        AnnotationLayer.delete_all
```
Comment (mr-sigma) [inline spec/controllers/annotation_lists_controller_spec.rb]: I believe this is what the DatabaseCleaner gem takes care of right? https://github.com/yale-web-technologies/mirador-annotations/blob/5f3b5c525c9daad3be040d329584b5b49a8b05d5/spec/rails_helper.rb#L68

**Pull request merged**